### PR TITLE
[Feature] Added Support for Client Side Event Filtering for Device Mode Destinations

### PR DESCRIPTION
**Pull request by desusai7**
## Description of the change

[Feature] Added Support for Client Side Event Filtering for Device Mode Destinations

Notion Link: [here](https://www.notion.so/rudderstacks/Event-Filtering-iOS-c1a4b5399cdf493faf6c1674a1f3e0d0)

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-ios/115)

### CODE REVIEW

File: Sources/Classes/RSEventFilteringPlugin.m [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+//  RSEventFilteringPlugin.m
+//  Rudder
+//
+//  Created by Desu Sai Venkat on 17/01/22.
+//  Copyright © 2022 Rudder Labs India Pvt Ltd. All rights reserved.
+//
+
+#import "RSEventFilteringPlugin.h"
+
+NSString* const DISABLE = @"disable";
```
Comment (pallabmaiti) [inline Sources/Classes/RSEventFilteringPlugin.m:11]: You should move these to `RSConstants` Class.

**Review by pallabmaiti — CHANGES_REQUESTED** [review]
I small change needed & we can go ahead.

File: Sources/Classes/RSEventFilteringPlugin.m (hunk 1 above)
Comment (desusai7) [reply Sources/Classes/RSEventFilteringPlugin.m:11]: Sure, will do

**Comment by pallabmaiti** [comment]
Let's follow the [process](https://www.notion.so/rudderstacks/Github-commit-messages-PR-creating-merging-process-c2c80cb466b843259e81f0a5e87c731e).

**Comment by desusai7** [comment]
@pallabmaiti, made the suggested changes, can you please take a look now

**Review by pallabmaiti — APPROVED** [review]

**Pull request merged**